Makefile

```diff
@@ -1,11 +1,29 @@
+.EXPORT_ALL_VARIABLES:
```

Comment:
My understanding of your approach is that all the environment variable are now shared across all the compilation stages. 
To avoid weird bugs, I'd rather keep the minimum set of environment variable set for every compilation stage to avoid weird bugs that could be hard to debug where an unrelated dependency is changing its behavior depending on an unrelated environment variable (this already happen to me in the past and it's not the kind of bugs that are nice to track)

Reply:
ok right